libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -807,15 +807,31 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	{
 		break;
 	}
-	case FunctionType::Kind::GasLeft:
+	case FunctionType::Kind::AddMod:
+	case FunctionType::Kind::MulMod:
 	{
-		define(_functionCall) << "gas()\n";
-		break;
+		solAssert(arguments.size() == 3 && parameterTypes.size() == 3, "");
+		Whiskers templ("if iszero(<var>) { revert(0, 0) }\n");
+		templ("var", expressionAsType(*arguments[2], *(parameterTypes[2])));
```

Comment:
If we use it like that, we convert the value twice. Maybe addmod and mulmod need their own case.

Reply:
ah.. I understand